File: selection.cpp
#include<iostream>

using namespace std;

void selection(int arr[], int n){
    for(int i =0;i<n;i++){
    int smallest = i;
    for(int j =i+1;j<n;j++){
        if(arr[j]  < arr[smallest]){
            smallest =j;
        }
    }
    swap(arr[i],arr[smallest]);
    }
}

void printarr(int arr[], int n){
    for(int i =0;i<n;i++){
        cout<<arr[i];
    }
}

int main(){
    int n =5;
    int arr[n] ={4,1,5,2,3};
    selection(arr,n);
    printarr(arr,n);
    return 0;
}

File: rev/selection.java

public class selection {
    
    public static void main(String[] args){
        int arr[] = {64,25,12,22,11};
        int n = arr.length; int min; int temp =0;

         for(int i =0;i<n;i++){
                min =i;
            for(int j =i+1;j<n;j++){
                if(arr[j] < arr[min]){
                    min = j;
                }
            }
                temp = arr[i];
                arr[i] = arr[min];
                arr[min]= temp;
            
         }

         for(int i =0;i<n;i++){
            System.out.print(arr[i]+" ");
         }
    }
}
